docker/client.go

```diff
@@ -67,8 +69,45 @@ func CheckSocket(socketPath string) bool {
 	return true
 }
 
+// IsWithinHostNetwork checks if a provided target is within the host container network
+func IsWithinHostNetwork(socketPath string, containerNameAsHostname bool, targetAddress string, targetPort int) (bool, error) {
```

Comment:
I think having this function is not needed because then you would have to get the list then do a for loop again to filter. In `ListContainers(socketPath string, enforceNetworkValidation bool, containerNameAsHostname bool) ([]Container, error)` you would already have everything you need to compile a final list.